### PR TITLE
feat(batch): add Batch labels job sample

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by Sita04** [comment]
@TetyanaYahodska What's the difference between this sample and the one in https://github.com/GoogleCloudPlatform/java-docs-samples/pull/9415?

**Comment by TetyanaYahodska** [comment]
@Sita04 This example demonstrates how to apply labels to a Job and #9415 do the same for a Runnable. We have three cases in the documentation https://cloud.google.com/batch/docs/organize-resources-using-labels#gcloud_1 and so three samples are created.

File: batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java [hunk 1]
```diff
@@ -58,12 +58,15 @@ public class CreateResourcesIT {
           + UUID.randomUUID().toString().substring(0, 7);
   private static final String CUSTOM_EVENT_NAME = "test-job"
           + UUID.randomUUID().toString().substring(0, 7);
+  private static final String BATCH_LABEL_JOB = "test-job"
```
Comment (Sita04) [inline batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java:61]:
```suggestion
  private static final String BATCH_LABEL_JOB = "test-job-label"
```

**Review by Sita04 — APPROVED** [review]
LGTM. One nit.

**Comment by Sita04** [comment]
Please follow https://www.conventionalcommits.org/en/v1.0.0/ for PR naming.

**Pull request merged**